lib/s3/models/BucketConfig.rb

```diff
@@ -172,6 +172,23 @@ def pretty_json
         }.reject { |k, v| v.nil? })
       end
 
+      # Public: deep sort an array of hashes or hash of arrays or array of arrays or hash of hashes or... you get the idea
+      #
+      #
+      # obj - The array or hash to deep sort
+      #
+      # Returns a deep sorted array of hashes or hash of arrays
+      def deep_sort_hash_array(obj)
+        case obj
+        when Array
+          obj.map!{|v| deep_sort_hash_array(v)}.sort_by!{|v| (v.to_s rescue nil) }
+        when Hash
+          obj = Hash[Hash[obj.map{|k,v| [deep_sort_hash_array(k),deep_sort_hash_array(v)]}].sort_by{|k,v| [(k.to_s rescue nil), (v.to_s rescue nil)]}]
```

Comment:
It there is a bad hash content it would not blow up very nicely. It is not 'needed' in this instance because I parse json before passing it into this function but those are in there in case someone else uses this. I do not care enough to push to leave it if you think that they should be removed.